util/s3/s3Helper.go

```diff
@@ -61,19 +63,56 @@ func loadS3Keys(keyFile string) (keys s3Keys, err error) {
 	return
 }
 
+func getS3UploadBucket() (bucketName string) {
+	bucketName, found := os.LookupEnv("S3_UPLOAD_BUCKET")
+	if !found {
+		bucketName = s3DefaultUploadBucket
+	}
+	return
+}
+
+func getS3ReleaseBucket() (bucketName string) {
+	bucketName, found := os.LookupEnv("S3_RELEASE_BUCKET")
+	if !found {
+		bucketName = algorandReleasesBucketName
+	}
+	return
+}
+
+func getS3Region() (region string) {
+	region, found := os.LookupEnv("S3_REGION")
+	if !found {
+		region = s3DefaultRegion
+	}
+	return
+}
+
+func (helper *Helper) UploadFileStream(filename string, reader io.Reader) error {
+	uploader := s3manager.NewUploader(helper.session)
+	_, err := uploader.Upload(&s3manager.UploadInput{
+		Bucket: aws.String(helper.bucket),
+		Key:    aws.String(filepath.Base(filename)),
+		Body:   reader,
+	})
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
 // MakeS3SessionForDownload returns an s3.Helper for the default algorand S3 bucket - for downloading
 func MakeS3SessionForDownload() (helper Helper, err error) {
 	// Create a session without credentials for the public algorand-releases bucket.
-	helper, err = makeS3Session(nil, algorandReleasesBucketName)
+	helper, err = makeS3Session(nil, getS3ReleaseBucket())
 	return
 }
 
 // MakeS3SessionForUpload returns an s3.Helper for the default algorand S3 bucket - for uploading
 func MakeS3SessionForUpload() (helper Helper, err error) {
 	awsID, _ := os.LookupEnv("AWS_ACCESS_KEY_ID")
 	awsKey, _ := os.LookupEnv("AWS_SECRET_ACCESS_KEY")
-	awsBucket := algorandBuildsBucketName
-	if awsID == "" || awsKey == "" || awsBucket == "" {
+	awsBucket := getS3UploadBucket()
+	if awsID == "" || awsKey == "" {
```

Comment:
still need to check for awsBucket == "" -- you can have a blank environment variable.

Reply:
OK, added a check for empty bucket name.